lib/chewy/type/adapter/object.rb

```diff
@@ -193,6 +193,16 @@ def load(ids, **options)
           end
         end
 
+        def import_count(*args)
+          collection = if args.first.empty? && @target.respond_to?(import_all_method)
+            @target.send(import_all_method)
+          else
+            args.flatten(1).compact
+          end
+
+          collection.count
```

Comment:
Won't it be a "bit" inefficient for huge data sources?

Reply:
Actually is hard to tell the advantages/disadvantages, because for huge data sources it could be particularly more interesting having a progress bar instead of just waiting without no feedback.

But yes, counting could take some time... But at least this is only calculated if someone asks for a progress bar.

WDYT? Keep as this or do you have other suggestion?

Reply:
Maybe, somebody will feel that calculations take too much time and then will turn off the progressbar.

Or does it makes sence to count the seconds of  `collection.count` executing? And we can stop the process via set `progressbar.title` to `Too many elements to output progressbar` in case of too long duration (like 2 seconds)?
@barthez @dalthon

Reply:
@barthez 
Should I change the approuch and just output the count of imported records to avoid `collection.count`? Or any other way?

@dalthon  
And I can try to use `Mutex` to synchronize the count of records to output during parallel import.

Reply:
I am totally ok with any of those approaches:
* Using the progress bar not as default so coping with the possibility of this performance cost happens with some kind of consent
* Changing this to a simple count of elements imported, of course handling the concurrency edge cases
* Having progress bar or count depending on env parameters, having simple count as default

Since @barthez doesn't seem to much fan of this introduced extra query for count, I guess the most promising of those 3 points are the 2nd and 3rd.

 Between those 2 last, I prefer the 2nd due to its simplicity and it would not add any new dependency to the project

Reply:
I will try to use thread to increment `progressbar.total` eventually during parallel import executing.